exhibitor-core/src/main/java/com/netflix/exhibitor/core/azure/AzureClientFactory.java

```diff
@@ -0,0 +1,13 @@
+package com.netflix.exhibitor.core.azure;
+
+public interface AzureClientFactory {
```

Comment:
there really is no need to abstract this interface (as it's already a specialized class)
if the `Client` class does have an interface, then it may be worth having an `Abstract Factory` that makes `Client`s:
```
    public abstract class ClientFactory {
        public Client makeClient(Credentials credentials);
    }
```
or something similar.